specs/IsSmartScreenRequired.md

```diff
@@ -0,0 +1,99 @@
+
+It is much easier to indicate if individual WebView2s require SmartScreen than to have an end developer explicitly manage if SmartScreen should be enabled as a whole, especially when its different sets of WebView2s in different processes (like Excel's set of WebView2s and Word's set of WebView2s) all sharing the same user data folder.
+In this document we describe the new setting.
+
+
+# Description
+You can use CoreWebView2Settings.IsSmartScreenRequired to control SmartScreen. SmartScreen is enabled or disabled per browser process, so all WebView2 applications sharing the same user data folder path also share SmartScreen being enabled or disabled.
+If CoreWebView2Setting.IsSmartScreenRequired is true for any CoreWebView2 in its associated CoreWebView2Environment, then SmartScreen is enabled. If CoreWebView2Setting.IsSmartScreenRequired is false for all CoreWebView2s in their CoreWebView2Environment, then SmartScreen is disabled.
+The default value for `IsSmartScreenRequired` is true if SmartScreen is enabled when the CoreWebView2 is created and false if SmartScreen is disabled when the CoreWebView2 is created. The value doesn't change if SmartScreen is enabled or disabled later.
+
+Changes to `IsSmartScreenRequired` take effect on the next navigation or download.
+
+
+# Examples
+```cpp
+// member variable
+wil::com_ptr<ICoreWebView2Settings> m_webViewSettings;
+
+// isLocalContent is TRUE if the page is navigated to content that is completely
+// app-provided, with no user-provided content or web-served content.
+// Note that we must update the property before navigating to the content.
+void SettingsComponent::UpdateSmartScreenRequirementBeforeNavigating(bool isLocalContent)
+{
+   wil::com_ptr<ICoreWebView2Settings11> coreWebView2Settings11;
+    coreWebView2Settings11 =
+                m_webViewSettings.try_query<ICoreWebView2Settings11>();
+    if(coreWebView2Settings11) 
+    {
+        CHECK_FAILURE(coreWebView2Settings11->put_IsSmartScreenRequired(!isLocalContent));
+    }
+}
+```
+
+```c#
+void ToggleSmartScreenRequired()
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsSmartScreenRequired = !settings.IsSmartScreenRequired;
+}
+```
+
+# Remarks
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```cpp
+[uuid(d667d3a7-c1b7-479f-8833-db7547df6687), object, pointer_default(unique)]
+interface ICoreWebView2Settings11 : ICoreWebView2Settings10 {
+  /// SmartScreen helps end users identify reported phishing and malware websites  
+  /// and also helps you make informed decisions about downloads. 
+  /// `IsSmartScreenRequired`  is used to control whether SmartScreen enabled or not. 
+  /// SmartScreen is enabled or disabled for all CoreWebView2s in a CoreWebView2Environment. 
```

Comment:
"In a CoreWebView2Environment" sounds like it's an instance of that object. But really this is profile wide?

Reply:
I'm not sure what you mean. Is 'all CoreWebView2s created with a particular CoreWebView2Environment' or 'all CoreWebView2s associated with a particular CoreWebView2Environment' better?

Yes SmartScreen is enabled or disabled for a CoreWebView2Environment.

Reply:
I think I'm generally confused about the scope of the property.

Reply:
SmartScreen is either on or off for the whole group of CoreWebView2s (that are all sharing the same CoreWebView2Environment, user data folder, and browser process). Each CoreWebView2 has a setting to say if it requires SmartScreen. If any in the group require it, then its turned on for the group. Is the documentation too confusing or the feature is too confusing?

Reply:
123

Reply:
@MikeHillberg 
Documentation has been updated to use the same user data folder for descriptions. Still confused?

Reply:
Thanx